### PR TITLE
adjust bld prometheus ext test for concurrent tests, cross namespace …

### PR DESCRIPTION
**Pull request by gabemontero**
…builds

Fixes https://github.com/openshift/origin/issues/17629

@openshift/sig-developer-experience fyi / ptal

### CODE REVIEW

File: test/extended/prometheus/prometheus_builds.go [hunk 1]
```diff
@@ -209,10 +215,10 @@ func labelsWeWant(sample *model.Sample, labels map[string]string) bool {
 }
 
 func valueWeWant(sample *model.Sample, tc metricTest) bool {
-	//NOTE - we could use SampleValue has an Equals func, but since SampleValue has no GreaterThan,
+	//NOTE - we could use SampleValue has an Equals func, but since SampleValue has no GreaterThanEqual,
 	// we have to go down the float64 compare anyway
-	if tc.greaterThan {
-		return float64(sample.Value) > tc.value
+	if tc.greaterThanEqual {
+		return float64(sample.Value) >= tc.value
```
Comment (bparees) [inline test/extended/prometheus/prometheus_builds.go]: wasn't the issue that we're getting a non-zero value when we expect zero?  how does this help that?

Comment (gabemontero) [reply test/extended/prometheus/prometheus_builds.go]: greaterThan was set to false in those cases, so we fell back to the == check below

if you noticed I set greaterThanEqual to true for cancelled / failed

Comment (bparees) [reply test/extended/prometheus/prometheus_builds.go]: got it.

**Comment by bparees** [comment]
lgtm pending passing tests.

**Comment by gabemontero** [comment]
tests passed @bparees .. note extended builds was skipped, but since this test is in the prometheus bucket as well, it got picked up by the conformance suite

File: test/extended/prometheus/prometheus_builds.go [hunk 2]
```diff
@@ -209,10 +215,10 @@ func labelsWeWant(sample *model.Sample, labels map[string]string) bool {
...
 	}
-	return float64(sample.Value) == tc.value
+	return float64(sample.Value) < tc.value
```
Comment (jim-minter) [inline test/extended/prometheus/prometheus_builds.go]: seems slightly odd that you're changing `==` to `<` here, not least given AFAICS there is no metricTest which doesn't have greaterThanEqual set, so this code path isn't actually ever used?

Comment (bparees) [reply test/extended/prometheus/prometheus_builds.go]: if you didn't explicitly request greaterThanEqual, then you're implicitly requesting less than... (at least under the new semantics introduced by this PR).

Comment (gabemontero) [reply test/extended/prometheus/prometheus_builds.go]: correct that the code path is not used currently; I wanted the code in place to facilitate usage of greaterThanEqual not being set in the future

and yes, given the precise values are not guarantee-able when this test is run in parallel / in conjunction with other tests, I wanted to have either "greater than or equal" or "less than or equal".

**Pull request merged**